src/common/types/uuid.cpp

```diff
@@ -0,0 +1,89 @@
+#include "common/types/uuid.h"
+
+namespace kuzu {
+namespace common {
+
+bool UUID::fromString(std::string str, int128_t& result) {
+    auto hex2Char = [](char ch) -> unsigned char {
+        if (ch >= '0' && ch <= '9') {
+            return ch - '0';
+        }
+        if (ch >= 'a' && ch <= 'f') {
+            return 10 + ch - 'a';
+        }
+        if (ch >= 'A' && ch <= 'F') {
+            return 10 + ch - 'A';
+        }
+        return 0;
+    };
+    auto isHex = [](char ch) -> bool {
+        return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+    };
+
+    if (str.empty()) {
+        return false;
+    }
+    int hasBraces = 0;
+    if (str.front() == '{') {
+        hasBraces = 1;
+    }
+    if (hasBraces && str.back() != '}') {
+        return false;
+    }
+
+    result.low = 0;
+    result.high = 0;
+    size_t count = 0;
+    for (size_t i = hasBraces; i < str.size() - hasBraces; ++i) {
+        if (str[i] == '-') {
+            continue;
+        }
+        if (count >= 32 || !isHex(str[i])) {
+            return false;
+        }
+        if (count >= 16) {
+            result.low = (result.low << 4) | hex2Char(str[i]);
+        } else {
+            result.high = (result.high << 4) | hex2Char(str[i]);
+        }
+        count++;
+    }
+    // Flip the first bit to make `order by uuid` same as `order by uuid::varchar`
+    result.high ^= (int64_t(1) << 63);
+    return count == 32;
+}
+
+void UUID::toString(int128_t input, char* buf) {
+    auto byteToHex = [](char byte_val, char* buf, uint64_t& pos) {
+        static char const HEX_DIGITS[] = "0123456789abcdef";
+        buf[pos++] = HEX_DIGITS[(byte_val >> 4) & 0xf];
+        buf[pos++] = HEX_DIGITS[byte_val & 0xf];
+    };
+
+    // Flip back before convert to string
+    int64_t high = input.high ^ (int64_t(1) << 63);
+    uint64_t pos = 0;
+    byteToHex(high >> 56 & 0xFF, buf, pos);
```

Comment:
It is better to declare those constants (56,48,40,32) as constexpr in the UUID class